File: packages/dialog/ios/Plugin/Classes/Results/PromptResult.swift
import Foundation
import Capacitor

@objc public class PromptResult: NSObject, Result {
    let value: String

    init(value: String) {
        self.value = value
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["value"] = value
        return result as AnyObject
    }
}

File: packages/dialog/android/src/main/java/io/capawesome/capacitorjs/plugins/dialog/classes/results/PromptResult.java
package io.capawesome.capacitorjs.plugins.dialog.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.dialog.interfaces.Result;

public class PromptResult implements Result {

    @NonNull
    private final String value;

    public PromptResult(@NonNull String value) {
        this.value = value;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("value", value);
        return result;
    }
}
